app/controllers/question_answer/comments_controller.rb

```diff
@@ -7,51 +7,45 @@ class CommentsController < ::ApplicationController
 
     def load_comments
       @guardian.ensure_can_see!(@post)
-      params.require(:post_number)
+      params.require(:last_comment_id)
 
       if @post.reply_to_post_number.present? && @post.post_number != 1
         raise Discourse::InvalidParameters
       end
 
-      posts =
-        Post
-          .where(
-            topic_id: @post.topic_id,
-            reply_to_post_number: @post.post_number,
-            post_type: Post.types[:regular],
-          )
-          .where("post_number > ?", params[:post_number])
-          .order(post_number: :asc)
+      comments =
+        QuestionAnswerComment
+          .where(topic_id: @post.topic_id, post_id: @post.id)
+          .where("id > ?", comments_params[:last_comment_id])
```

Comment:
Just to confirm, this `comments` are comments for a post, right? It would be nice for a better name that's scoped to the query, or even a named scope maybe. I don't get the need for the `last_comment_id` clause.

Reply:
Will renaming the controller action to `load_more_comments` make it clearer? This action is used to load more comments in the mini comments stream.